agent/lib/kontena/rpc_client.rb

```diff
@@ -88,7 +88,7 @@ def request_id
       id = rand(REQUEST_ID_RANGE)
 
       while @requests.has_key?(id)
-        sleep 0.001
```

Comment:
This is a mystery to me, I don't know what happened to break this with Ruby 2.4, and Celluloid itself should not have updated.... somehow the 1ms sleep is causing the call to `subject.handle_response` in the specs to never actually happen.